### ISSUE TITLE
UnityParams java.util.ConcurrentModificationException

### ISSUE THREAD

**Issue by kicholen** [facebook/facebook-sdk-for-unity#71]
Hello, 
There seems to be problem with parsing UnityParams, happened 3 times so far only on Android devices version 5.1.1. We do not use FB.Mobile.FetchDeferredAppLinkData at all in our application. Here is info and stack trace:

Facebook Sdk: 7.8.0
Platform: android
Version: N/A
Date: 2017-03-23T04:50:40Z
OS Version: 5.1.1
Device: A37f

Fatal Exception: java.lang.Error: FATAL EXCEPTION [AsyncTask 4]
Unity version     : 5.5.1f1
Device model      : OPPO A37f

`Caused by java.util.ConcurrentModificationException
       at java.util.HashMap$HashIterator.nextEntry(HashMap.java:787)
       at java.util.HashMap$EntryIterator.next(HashMap.java:824)
       at java.util.HashMap$EntryIterator.next(HashMap.java:822)
       at org.json.JSONObject.<init>(JSONObject.java:129)
       at com.facebook.unity.UnityParams.<init>(UnityParams.java:47)
       at com.facebook.unity.UnityMessage.send(UnityMessage.java:58)
       at com.facebook.unity.FB$4.onDeferredAppLinkDataFetched(FB.java:348)
       at com.facebook.applinks.AppLinkData.fetchDeferredAppLinkFromServer(AppLinkData.java:239)
       at com.facebook.applinks.AppLinkData.access$000(AppLinkData.java:50)
       at com.facebook.applinks.AppLinkData$1.run(AppLinkData.java:148)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1115)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:590)
       at java.lang.Thread.run(Thread.java:818)`

I can provide more info or stacktrace from diffrent device if needed.

**Comment by hu9o** [comment]
+1, ran into the same issue

**Comment by aaronhoogstraten** [comment]
I also have encountered this issue for about 15 users:
```
Fatal Exception: java.lang.Error: FATAL EXCEPTION [main]
Unity version     : 5.6.3p3
Device model      : samsung SM-G935F
Device fingerprint: samsung/hero2ltexx/hero2lte:7.0/XXXX/XXXXXXX:user/release-keys

Caused by java.util.ConcurrentModificationException
       at java.util.HashMap$HashIterator.nextEntry(HashMap.java:851)
       at java.util.HashMap$EntryIterator.next(HashMap.java:891)
       at java.util.HashMap$EntryIterator.next(HashMap.java:890)
       at org.json.JSONObject.<init>(JSONObject.java:129)
       at com.facebook.unity.UnityParams.<init>(UnityParams.java:47)
       at com.facebook.unity.UnityMessage.send(UnityMessage.java:58)
       at com.facebook.unity.FB$3.OnTokenRefreshed(FB.java:333)
       at com.facebook.AccessTokenManager$4.onBatchCompleted(AccessTokenManager.java:325)
       at com.facebook.GraphRequest$5.run(GraphRequest.java:1384)
       at android.os.Handler.handleCallback(Handler.java:751)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:154)
       at android.app.ActivityThread.main(ActivityThread.java:6692)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)
```

**Comment by brandesign** [comment]
We run into the same issue with SDK version 7.10.1

> java.lang.Error: FATAL EXCEPTION [AsyncTask #5]
Unity version     : 5.6.5p1
Device model      : samsung SM-G930F
Device fingerprint: samsung/heroltexx/herolte:7.0/NRD90M/G930FXXS2DRAA:user/release-keys
Caused by
  at java.util.HashMap$HashIterator.nextEntry (HashMap.java:851)
  at java.util.HashMap$EntryIterator.next (HashMap.java:891)
  at java.util.HashMap$EntryIterator.next (HashMap.java:890)
  at org.json.JSONObject.<init> (JSONObject.java:129)
  at com.facebook.unity.UnityParams.<init> (UnityParams.java:47)
  at com.facebook.unity.UnityMessage.send (UnityMessage.java:58)
  at com.facebook.unity.FB$4.onDeferredAppLinkDataFetched (FB.java:366)
  at com.facebook.applinks.AppLinkData.fetchDeferredAppLinkFromServer (AppLinkData.java:242)
  at com.facebook.applinks.AppLinkData.access$000 (AppLinkData.java:50)
  at com.facebook.applinks.AppLinkData$1.run (AppLinkData.java:148)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1133)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:607)
  at java.lang.Thread.run (Thread.java:762)

**Comment by kicholen** [comment]
I think it's fixed in 7.11.0 by removing:
https://github.com/facebook/facebook-sdk-for-unity/commit/07767e204705b7e2452c67108d39b0959926ebf7#diff-34c240c792c7c707e531f8dd120704d4L360
